### DISCUSSION TITLE
The start option of xpraclient is not available. Please help me

### DISCUSSION THREAD

**Discussion by hhc8888888** [Xpra-org/xpra#3805] [Q&A] [answered]
Hello everyone, I have been trying to use xpra recently. There is now a small problem. I started the session using xpra 3.1.3 on centos 7.9 and started xterm. Then I connected to the session using xpraclient 3.1.3 on Windows. At this point, I find that the start option on the client is not available and prompt "This server does not provide start menu data". How can I solve this problem? I expect xpraclients located on windows to be able to choose from the start option for application startup. Can you help me?
![image](https://user-images.githubusercontent.com/127097977/227733341-c8d99fda-0b9e-4c8d-ab0a-ff374971bd54.png)

**Comment by totaam** [comment, marked as answer]
> Then I connected to the session using xpra client 3.1.3 on Windows

Please try the latest version.
At time of writing, this is either 3.1.4 from the 3.1.x branch, or 4.4.4 from the 4.4.x branch.

> "This server does not provide start menu data"

The server does not have the required libraries installed for exposing the start menu data to the client.
You should be able to run the menu helper:
https://github.com/Xpra-org/xpra/blob/v3.1.x/xpra/platform/xposix/xdg_helper.py

Perhaps CentOS 7.x doesn't have `python2-pyxdg` ?
If so then I would recommend moving to a more recent distribution.

**Reply by hhc8888888** [reply]
Oh, My God
I can really use it. The problem that has been bothering me for two days has been solved through your contempt!
Now I have a new idea. I want to make the "start" option only select a certain application, such as "xterm". Can you tell me how to do this?

**Reply by hhc8888888** [reply]
![image](https://user-images.githubusercontent.com/127097977/227767699-d7b5392b-8633-4340-9737-7fd37e5b0fe7.png)

**Reply by hhc8888888** [reply]
Like this
![image](https://user-images.githubusercontent.com/127097977/227768007-1017c2fa-658d-4fcc-bafd-ba1119fd3611.png)

**Reply by totaam** [reply]
> has been solved through your contempt!

Are you sure you meant to use the word "contempt" here?

**Comment by totaam** [comment]
As for selecting applications, you can copy your selection from `/usr/share/applications` into a directory of your choice, then point to it when starting the server with:
```shell
xpra start --env=XPRA_MENU_LOAD_APPLICATIONS=/your/directory
```
And this should populate the start menu with the contents of your directory.

**Reply by hhc8888888** [reply]
Thank you, my friend on the other side of the globe, for helping me solve my problems. "xpra" is really great software, and I applaud you

**Reply by hhc8888888** [reply]
![image](https://user-images.githubusercontent.com/127097977/227776317-be4c9c68-aa5a-4eae-8b2a-fdddd7e2926a.png)
As you said, I only placed "xterm" in the "/usr/share/hhc_applications" directory. Why are there other applications available in "start"?

**Reply by hhc8888888** [reply]
I want to specify the app displayed in "start". For now, I just want 'xtrem' to appear in the 'start' option.Can you help me. Please, buddy.

**Reply by totaam** [reply]
The option I gave you works for recent versions, not for v3.1.x

I strongly recommend that you upgrade to something more recent.

If you insist on running the old version, you may be able to to set `XDG_DATA_DIRS` instead.
But this may well have undesirable side-effects.
